src/psyclone/generator.py

```diff
@@ -498,3 +538,49 @@ def main(args):
             psy_file.write(psy_str)
     else:
         print(f"Generated psy layer code:\n{psy_str}")
+
+
+def check_fp2_tree(fp2_tree, filename):
```

Comment:
Please add comments saying that this code will be removed.

Reply:
I'm not quite sure whether we're talking at cross purposes here. To be specific, is `check_fp2_tree` intended to be a permanent addition? If so, you can ignore my original comment.

Reply:
The check is permanent but I've modified the implementation so that the check is now done in the PSyIR rather than the fparser2 tree. To do this I needed to modify the `add_builtins_use` function to work with multiple programs/modules.

Reply:
OK. Thanks for improving it to use the PSyIR.